python/src/trezorlib/transport/udp.py

```diff
@@ -142,14 +142,17 @@ def write_chunk(self, chunk: bytes) -> None:
         LOG.log(DUMP_PACKETS, f"sending packet: {chunk.hex()}")
         self.socket.sendall(chunk)
 
-    def read_chunk(self) -> bytes:
+    def read_chunk(self, timeout: float | None = None) -> bytes:
         assert self.socket is not None
+        start = time.time()
         while True:
             try:
                 chunk = self.socket.recv(64)
                 break
             except socket.timeout:
-                continue
+                if timeout is not None and time.time() - start > timeout:
+                    raise Timeout("Timeout reading UDP packet")
+                time.sleep(0.001)
```

Comment:
you were probably right about this being useless; the sleep technically happens on the `recv()` itself

Reply:
Sounds good, will remove.

Reply:
f20249192bb8e4b789c484ba12fda7d382e740b3